vignettes/updraft-intro.Rmd

```diff
@@ -0,0 +1,240 @@
+---
+title: "Build and Execute Modular Workflows in R"
+output: 
+  rmarkdown::html_vignette:
+    toc: true
+    fig_caption: yes
+vignette: >
+  %\VignetteIndexEntry{Build and Execute Modular Workflows in R}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+
+library(updraft)
+```
+
+
+`updraft` is an R package designed to simplify building and executing of "workflows", or a sequence of modular pieces of code with interdependencies that must be run. This enables you to decompose your code into focused, modular components ("modules") and then execute without worrying about managing the dependencies between each. Additionally, this package enables you to:
+
+- automatically parallelize the execution of your workflow
+- visualize your workflow 
+- save and load your workflow from file
+
+Below is an introduction to the concepts underlying `updraft`, along with some examples of how it can be leveraged.
+
+## Constructing and Running a Workflow
+
+### Module
+
+The `Module` is the basic building block of `updraft`. A `Workflow` is merely a set of `Module`s to be executed. The smaller and more single-purpose these `Modules` are, the better. There are two existing implementations, both aligning to the interface in `ModuleInterface`.
+
+#### Package Function Module
+
+The `PackageFunctionModule` object is for when you merely want to execute a function that exists in some package. For example, 
+
+```{r packageFunctionModule, eval=TRUE}
+
+pasteInputs <- PackageFunctionModule$new(name = "pasteInputs"
+                                         , fun = "paste0"
+                                         , package = "base")
+
+```
+To execute this simple module outside of a workflow, we can do so as follows:
+
+```{r execModule, eval=TRUE}
+
+pasteInputs$startExecution(list("hello, ", "world!"))
+
+# to see the output
+pasteInputs$getOutput()
+
+```
+
+The node encapsulates the assigned function and allows you to access its output value, along with metadata related to the calculation.
+
+#### Custom Function
+
+More typically, you'll want to do more complicated operations than vanilla package function calls. This is accomplished with the `CustomFunctionModule` class.
+
+For example, let's say you want to apply a custom function to inputs. You can easily define the function and instantiate a `CustomFunctionModule` using it, as shown below:
+
+```{r customFunctionModule, eval=TRUE}
+
+myFunc <- function(a, b, c) {
+    return(a + 2*(floor(b/c)))
+}
+
+myModule <- CustomFunctionModule$new(name = "myFunc"
+                                     , fun = myFunc)
+
+
+```
+
+As with the `PackageFunctionModule` we can merely execute this, but there is even more we can do.
```

Comment:
could you expand this text? You say "there is more we can do" and then there isn't an explanation of what the code below actually does or why I should think "wow that is cool!"